spark/src/test/scala/geotrellis/spark/io/hadoop/HadoopTileFeatureSpatialSpec.scala

```diff
@@ -0,0 +1,45 @@
+package geotrellis.spark.io.hadoop
+
+import geotrellis.raster.{Tile, TileFeature}
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.index._
+import geotrellis.spark.testfiles.TestTileFeatureFiles
+
+
+class HadoopTileFeatureSpatialSpec
+  extends PersistenceSpec[SpatialKey, TileFeature[Tile, Tile], TileLayerMetadata[SpatialKey]]
+    with SpatialKeyIndexMethods
+    with TestEnvironment
+    with TestTileFeatureFiles
+    with AllOnesTestTileFeatureTests {
+
+  lazy val reader = HadoopLayerReader(outputLocal)
+  lazy val writer = HadoopLayerWriter(outputLocal)
+  lazy val deleter = HadoopLayerDeleter(outputLocal)
+  lazy val copier = HadoopLayerCopier(outputLocal)
+  lazy val mover  = HadoopLayerMover(outputLocal)
+  lazy val reindexer = HadoopLayerReindexer(outputLocal)
+  lazy val updater = HadoopLayerUpdater(outputLocal)
+  lazy val tiles = HadoopValueReader(outputLocal)
+  lazy val sample = AllOnesTestFile
+
+  describe("HDFS layer names") {
+    it("should handle layer names with spaces") {
+      val layer = AllOnesTestFile
+      val layerId = LayerId("Some layer", 10)
+
+      writer.write[SpatialKey, TileFeature[Tile, Tile], TileLayerMetadata[SpatialKey]](layerId, layer, ZCurveKeyIndexMethod)
+      val backin = reader.read[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](layerId)
+    }
+
+    it("should fail gracefully with colon in name") {
```

Comment:
Oh I see, this was copied from the non-feature tile reader. I don't think we have to test this twice.